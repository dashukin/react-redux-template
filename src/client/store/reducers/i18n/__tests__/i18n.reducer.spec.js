import i18nReducer, {
  initialState,
} from '../i18n.reducer';

import i18nConstants from '../i18n.constants';

const {
  I18N_INIT_SUCCESS,
} = i18nConstants;

describe('i18n reducer', () => {
  describe('initial state', () => {
    it('should match snapshot', () => {
      expect(initialState).toMatchSnapshot();
    });
  });

  it(`should update code on ${I18N_INIT_SUCCESS} action`, () => {
    const mockI18nCode = 'test-code';
    const defaultState = i18nReducer(undefined, { type: undefined });

    expect(defaultState).toEqual(initialState);

    const updatedState = i18nReducer(defaultState, {
      type: I18N_INIT_SUCCESS,
      payload: {
        code: mockI18nCode,
      },
    });

    const expectedState = {
      ...defaultState,
      code: mockI18nCode,
    };
    expect(updatedState).toEqual(expectedState);
  });
});
