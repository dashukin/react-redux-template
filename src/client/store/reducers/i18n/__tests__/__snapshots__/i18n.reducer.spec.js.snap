// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`i18n reducer initial state should match snapshot 1`] = `
Object {
  "code": undefined,
}
`;
